lib/xrp/XrpClientAdapter.js

```diff
@@ -0,0 +1,126 @@
+const xrpl = require('xrpl');
+
+// Array elements should be lower-case
+const SUPPORTED_TRANSACTION_TYPES = new Set(['payment']);
+// https://xrpl.org/docs/references/protocol/transactions/transaction-results
+const SUPPORTED_TRANSACTION_RESULT_CODES = {
+  tes: {
+    SUCCESS: 'tesSUCCESS'
+  }
+};
+
+/**
+ * This adapter is used to adapt the new XRP client provided by the xrpl v2 dependency
+ * so that it will behave like the old XRP client provided by ripple-lib v1
+ * 
+ * Migration guide: https://xrpl.org/docs/references/xrpljs2-migration-guide
+ * ripple-lib ref: https://github.com/XRPLF/xrpl.js/blob/1.x/docs/index.md
+ */
+class XrpClientAdapter extends xrpl.Client {
+  async getServerInfo() {
+    return await this.request({
+      id: 1,
+      command: 'server_info'
+    });
+  }
+
+  async getLedger({ ledgerVersion }) {
+    return await this.request({
+      command: 'ledger',
+      ledger_index: ledgerVersion
+    });
+  }
+
+  /**
+     * Retrieves transactions based on the provided parameters.
+     *
+     * @param {string} acceptanceAddress - The address of the account to get transactions for.
+     * @param {Object} options - The options for retrieving transactions.
+     * @param {number} [options.minLedgerVersion] - Return only transactions in this ledger version or higher.
+     * @param {number} [options.maxLedgerVersion] - Return only transactions in this ledger version or lower.
+     * @param {Array<string>} [options.types] - Only return transactions of the specified Transaction Types (see SUPPORTED_TRANSACTION_TYPES).
+     * @param {boolean} [options.initiated] - If true, return only transactions initiated by the account specified by acceptanceAddress. If false, return only transactions NOT initiated by the account specified by acceptanceAddress.
+     * @param {boolean} [options.includeRawTransactions] - Include raw transaction data. For advanced users; exercise caution when interpreting this data.
+     * @param {boolean} [options.excludeFailures] - If true, the result omits transactions that did not succeed.
+     * @returns {Promise<Array>} A promise that resolves to an array of transactions.
+     */
+  async getTransactions(acceptanceAddress, {
+    minLedgerVersion,
+    maxLedgerVersion,
+    types,
+    initiated,
+    includeRawTransactions,
+    excludeFailures
+  }) {
+    /**
+         * Behavior defaults to 'true', but this error is to document that 'includeRawTransactions: false' is NOT supported
+         * Truthiness is not sufficient for this check - it must explicitly be an equality check, & strict equality is prefered
+         */
+    if (includeRawTransactions === false) {
+      throw new Error('"includeRawTransactions: false" not supported');
+    }
+
+    /**
+         * Filtering constants with defaults
+         */
+    const TYPES = Array.isArray(types) ? 
```

Comment:
Actually I'm thinking on this more now... because if getTransactions is called with an empty array, then having the behavior default to include Payments doesn't make much sense...

That would be a strange way to use this method though - you'd be essentially filtering out all transactions...

I'm probably overthinking it.